compiler/AST/ForLoop.cpp

```diff
@@ -98,6 +119,21 @@ static void tryToReplaceWithDirectRangeIterator(Expr* iteratorExpr)
       Expr* high = range->get(2)->copy();
       iteratorExpr->replace(new CallExpr("chpl_direct_range_iter", low, high, stride));
     }
+
+    // replace a counted, low bounded range with unit stride with an equivalent
+    // direct range iter. e.g. replace:
+    //
+    //   `low..#count` (which is equivalent to `low..low+count-1`)
+    //
+    // with:
+    //
+    //   `chpl_direct_range_iter(low, low+count-1, 1)`
+    else if (count && range && range->isNamed("chpl_build_low_bounded_range"))
+    {
+      Expr* low = range->get(2)->copy();
```

Comment:
The magic number "2" here might deserve a quick comment.  Intuitively, I would've guessed the low bound was argument 1.  Looking at the parser, I see that that's the enum argument (?).  Could be something short and sweet like //arg 1 = BoundedRangeType.boundedLow, arg2 = low bound.

Having typed that, though, now that chpl_build_partially_bounded_range() has been split into two cases, why require the two cases to take the enums anymore?  It seems like build_low_bounded and build_high_bounded will always take the exact same enum value in all cases (?).